linkerd/app/inbound/fuzz/fuzz_targets/fuzz_target_1.rs

```diff
@@ -0,0 +1,24 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+use libfuzzer_sys::arbitrary::Arbitrary;
+extern crate hyper;
+
+#[derive(Debug, Arbitrary)]
+struct TransportHeaderSpec {
+        data: Vec<u8>,
+        h1: Vec<u8>,
+        h2: Vec<u8>,
+        protocol: bool,
+}
+
+//fuzz_target!(|data: &[u8]| {
+fuzz_target!(|inp: TransportHeaderSpec| {
+    if let Ok(s) = std::str::from_utf8(&inp.data[..]) {
```

Comment:
Since this is used as a URI, do we want to validate that the string is a valid URI? Or is the intent to test invalid URIs?

Reply:
I would first go for invalid URIs too, then, if we observe any issues with it e.g. false positives we can fix it. In my test this seems to work well.